Archives/ClmFile.cpp

```diff
@@ -291,7 +292,7 @@ namespace Archives
 		for (std::size_t i = 0; i < internalNames.size(); ++i)
 		{
 			// Copy the filename into the entry
-			strncpy((char*)&indexEntries[i].fileName, internalNames[i].c_str(), 8);
+			std::memcpy(indexEntries[i].fileName.data(), internalNames[i].data(), sizeof(IndexEntry::fileName));
```

Comment:
I think I might still use [`stdncpy`](https://en.cppreference.com/w/cpp/string/byte/strncpy) here. It will not attempt to read past the end of the input string. Additionally, it null pads the buffer up to the stated size. That means no uninitialized data laying around in the index table.

Reply:
For the record, I do rather like the use of `sizeof` here. That was a nice touch.